### PR TITLE
Cell methods

### PR DESCRIPTION
**Pull request by benjwadams**
Adds fixes for cell methods requested for #509.  Additional test coverage.

### CODE REVIEW

**Comment by benjwadams** [comment]
The package dependency I added doesn't have a corresponding installation candidate in Conda for Python 3.4

**Comment by benjwadams** [comment]
@ocefpaf, it's been a while since I've run into one of these conda build failures.  I'd need to rebuild the package for the affected versions and upload to conda-forge, correct?

**Comment by ocefpaf** [comment]
@benjwadams conda-forge dropped support for Python 3.4 and I suggest we do the same here.

See https://github.com/ioos/compliance-checker/pull/518

File: compliance_checker/tests/test_cf.py [hunk 1]
```diff
@@ -967,6 +962,29 @@ def test_check_cell_methods(self):
...
+        nc_obj = MockTimeSeries()
+        nc_obj.createVariable('temperature', 'd', ('time',))
+
+        temp = nc_obj.variables['temperature']
+        temp.cell_methods = 'lat: lon: mean depth: mean (interval: 20 meters)'
+        results = self.cf.check_cell_methods(nc_obj)
+        scored, out_of, messages = self.get_results(results)
+        result_dict = {result.name: result for result in results}
+        modifier_results = result_dict[u'§7.3.3 temperature has valid cell_methods modifiers']
+        self.assertTrue(sub(*modifier_results.value[::-1]) == 0)
```
Comment (Bobfrat) [inline compliance_checker/tests/test_cf.py]: @benjwadams can you add a comment here?

**Comment by Bobfrat** [comment]
@benjwadams just to summarize, this PR adds a check of valid units in a cell_methods string.

For example, in the issue example file the following would be flagged due to the misspelling of 'minutes'.
```
hurs:cell_methods = "time: mean (interval: 20 mintues)" ;
```

**Comment by benjwadams** [comment]
@Bobfrat, yes and it also checks that the number specified after the interval is in fact parsable to a numeric type.

File: compliance_checker/tests/test_cf.py (hunk 1 above)
Comment (benjwadams) [reply compliance_checker/tests/test_cf.py]: Superceded by a simpler scoring check.

**Pull request merged**